sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs

```diff
@@ -12,6 +15,12 @@ namespace System.ClientModel.Primitives;
 /// </summary>
 public static class ModelReaderWriter
 {
+    private static readonly HashSet<Type> s_supportedCollectionTypes =
```

Comment:
Do we expect the supported types to grow over time? With just a few elements an iterative lookup can be faster than the hash.

Reply:
I think it would be easy to add things like Collection, SortedList, etc if we wanted to and this just centralized the checking to make it simpler than a series of if/else statements.

Still in discussion on which types to support for output

Reply:
How does this work for `ReadOnlyCollection<T>` (or whatever type we instantiate for an `IReadOnlyList<T>` output collection)?  Do we need to update this with every possible collection type we return?  How does it work for internal shared source types like `ChangeTrackingList<T>`?

Reply:
For output the idea is we are limiting you to a set collection types.  Right now if you wanted a ChangeTrackingList<T> you would do this

```c#
ChangeTrackingList<MyType> list = new(ModelReaderWriter.Read<List<MyType>>(json));
```

For ReadOnlycollection<T> you would do
```c#
ReadOnlycollection<MyType> readonlyList = ModelReaderWriter.Read<List<MyType>>(json).AsReadOnly();
```

We can add support to do this internally, but we would need to do something special for each of these that we want to add so its a balance of how many types we think should be in this supported list.

Reply:
the restriction no longer exists